openmcyclus/depletion.py

```diff
@@ -3,156 +3,192 @@
             location as the other files for the OpenMC model
         timesteps: int
             number of time steps to perform depletion. It is assumed that
-            the number given is in units of months. This value is 
-            multiplied by 30 to give a timestep in days. 
+            the number given is in units of months. This value is
+            multiplied by 30 to give a timestep in days.
         power: float
             power output of the reactor, assumed in MWe.
         conversion_factor: float
             conversion factor to go from MWe to MWth. Default value of 3,
             representing a thermal efficiency of 1/3
         '''
-        self.path = pathlib.Path(path)
-        self.prototype = prototype
         self.chain_file = chain_file
         self.timesteps = timesteps
         self.power = power
         self.conversion_factor = conversion_factor
 
-    def read_model(self):
+    def read_model(self, path):
         '''
         Read the .xml files in the defined path to create an OpenMC
         model in the python API. The files are assumed to be named
         geometry.xml, materials.xml, settings.xml, and (optionally)
         tallies.xml
 
+        Parameters:
+        -----------
+        path: str
+            path of directory holding the files for/from OpenMC
+
         Returns:
         ---------
         model: openmc.model.model object
             OpenMC model of reactor geometry.
         '''
-        model_kwargs = {"geometry": (self.path / "geometry.xml"),
-                        "materials": (self.path / "materials.xml"),
-                        "settings": (self.path / "settings.xml")}
-        tallies_path = self.path / "tallies.xml"
-        if tallies_path.exists():
-            model_kwargs["tallies"] = tallies_path
+        model_kwargs = {"geometry": path + "geometry.xml",
+                        "materials": path + "materials.xml",
+                        "settings": path + "settings.xml"}
         model = openmc.model.model.Model.from_xml(**model_kwargs)
         return model
 
-    def read_microxs(self):
+    def update_materials(self, comp_list, path):
+        '''
+        Read in the material compositions of the fuel assemblies present
+        in the reactor to be transmuted. Then modify the composition of
+        the pre-defined materials to match the compositions from
+        Cyclus.
+
+
+        Parameters:
+        -----------
+        comp_list: list of dicts
+            list of the fresh fuel compositions present in the core
+            at the calling of the transmute function.
+        path: str
+            path of directory holding the files for/from OpenMC
+
+        Returns:
+        --------
+        material_ids: list of strs
+            material id numbers for the OpenMC model
+
+        Outputs:
+        --------
+        materials.xml: file
+            updated XML for OpenMC with new compositions
+
```

Comment:
Since this function isn't a generator and doesn't "yield" anything, and Numpydoc doesn't have an `Outputs` header, I would simply remove this section and describe the file it writes to in the main docstring.
```suggestion
```